### PR TITLE
Issue/1502

### PR DESCRIPTION
**Pull request by canstudios-louisem**
* Bring in Kineo tests
* Tests now always start with a clean DB
* Upgrade should as Kineo tests needed it
* Fix tests broken from changes and should upgrade

#1502

### CODE REVIEW

File: lib/tenantmanager.js [hunk 1]
```diff
@@ -489,18 +490,15 @@ exports = module.exports = {
...
         }
-
-        if (results && results.length === 1) {
-          return self.updateTenant({ '_id': results[0]._id }, { _isDeleted: true }, callback);
+        if (!result) {
+          return callback(new Error('No matching tenant record found'));
         }
-
-        return callback(new Error('No matching tenant record found'));
+        return self.updateTenant({ '_id': results[0]._id }, { _isDeleted: true }, callback);
```
Comment (tomgreenfield) [inline lib/tenantmanager.js]: `results` is no longer defined?

**Comment by taylortom** [comment]
Tests should be working smoothly now

**Comment by canstudios-nicolaw** [comment]
The tests are all passing for me, I have noticed I get the following output when I run them

`error: [07 Aug 2017 10:56:45 +01:00] test foo=bar`

I just wanted to check this is something we are expecting/aware of?

**Comment by taylortom** [comment]
@canstudios-nicolaw that's because of [this test](https://github.com/adaptlearning/adapt_authoring/blob/issue/1502/test/lib.logger.js#L10-L17). We can set up the logger to mute all messages during the tests if we want?

**Comment by canstudios-nicolaw** [comment]
Ok I am happy as long as we understand where it is coming from :) It feels a bit dangerous to mute all messages in case we miss something in future.

**Comment by taylortom** [comment]
Have updated the test script to use the grunt 'test' task, thus removing the need for a makefile.

File: Gruntfile.js [hunk 2]
```diff
@@ -349,6 +349,6 @@ module.exports = function(grunt) {
     }
   });
 
-  grunt.registerTask('test', ['mochaTest', 'casperjs']);
+  grunt.registerTask('test', ['mochaTest'/*, 'casperjs'*/]);
```
Comment (canstudios-louisem) [inline Gruntfile.js]: Should we update the package.json's test command as it curently does make test not grunt test. Im not sure how much it matters really at Can we all use php storm's mocha test runner to run unit tests.

File: Gruntfile.js [hunk 3]
```diff
@@ -357,7 +349,6 @@ module.exports = function(grunt) {
     }
   });
 
+  grunt.registerTask('test', ['mochaTest'/*, 'casperjs'*/]);
```
Comment (canstudios-louisem) [inline Gruntfile.js]: Should we update the package.json's test command as it curently does make test not grunt test. Im not sure how much it matters really at Can we all use php storm's mocha test runner to run unit tests.

Comment (taylortom) [inline Gruntfile.js]: The latest package.json should use `grunt test`

**Comment by taylortom** [comment]
@canstudios-louisem you're right, I just knew if I waited to do a separate PR, I'd likely forget 😅

**Comment by canstudios-louisem** [comment]
that's why i added a reminder months a go :) https://github.com/adaptlearning/adapt_authoring/issues/1649

File: Gruntfile.js (hunk 3 above)
Comment (canstudios-louisem) [inline Gruntfile.js]: Ah nice for some reason i missed you had changed it.

**Pull request merged**